src/ansi-c/c_typecheck_expr.cpp

```diff
@@ -3254,10 +3140,8 @@ void c_typecheck_baset::typecheck_arithmetic_pointer(const exprt &expr)
 
 void c_typecheck_baset::typecheck_expr_pointer_arithmetic(exprt &expr)
 {
-  assert(expr.operands().size()==2);
-
-  exprt &op0=expr.op0();
-  exprt &op1=expr.op1();
+  exprt &op0 = to_binary_expr(expr).op0();
+  exprt &op1 = to_binary_expr(expr).op1();
```

Comment:
I think we should do `to_binary_expr` just once.

Reply:
done